File: solutions_5648941810974720_1/Python/pvncad/digit.py
import sys

from collections import defaultdict

T = int(raw_input())

for tc in xrange(1, T + 1):
    line = raw_input().strip()
    cc = defaultdict(lambda : 0)

    for c in line:
        cc[c] += 1

    dc = [0] * 10

    dc[0] = cc['Z']
    dc[2] = cc['W']
    dc[6] = cc['X']
    dc[8] = cc['G']
    dc[4] = cc['U']

    for (d, s) in [ (0, "ZERO"), (2, "TWO"), (4, "FOUR"), (6, "SIX"), (8, "EIGHT")]:
        for x in s:
            cc[x] -= dc[d]

    dc[3] = cc['H']
    dc[5] = cc['F']
    dc[1] = cc['O']
    dc[7] = cc['S']

    for (d, s) in [(3, "THREE"), (5, "FIVE"), (1, "ONE"), (7, "SEVEN")]:
        for x in s:
            cc[x] -= dc[d]

    dc[9] = cc['I']

    ans = ""

    for d in xrange(10):
        if dc[d]:
            ans += ("%d" % d) * dc[d]

    print "Case #%d: %s" % (tc, ans)

File: solutions_5648941810974720_0/java/trainsick/Digit.java

import java.util.Scanner;
import java.io.PrintStream;

public class Digit {
    
    public static void main(String[] args) {
        new Digit().main2();
    }
    
    Scanner sc=new Scanner(System.in);
    PrintStream out=System.out;
    
    void main2() {
        int i,T;
        T=sc.nextInt();
        for (i=1;i<=T;i++) {
            out.printf("Case #%d: ",i);
            problem();
        }
    }

    int[] count;
    int[] ans;

    
    int prune(int n, String s) {
        for (char c : s.toCharArray()) {
            count[c]-=n;
        }
        return n;
    }
    
    void isempty() {
        for (int i=48;i<58;i++) {
            if (count[i]!=0) out.println("NOT EMPTY");
        }
    }
    
    void problem() {
        String letters=sc.next();
        StringBuilder digits=new StringBuilder();
        count=new int[128];
        ans=new int[10];
        for (char c : letters.toCharArray()) {
            if (c<128) count[c]++;
        }
        ans[0]=prune(count['Z'],"ZERO");
        ans[2]=prune(count['W'],"TWO");
        ans[4]=prune(count['U'],"FOUR");
        ans[6]=prune(count['X'],"SIX");
        ans[8]=prune(count['G'],"EIGHT");
        ans[1]=prune(count['O'],"ONE");
        ans[3]=prune(count['H'],"THREE");
        ans[5]=prune(count['F'],"FIVE");
        ans[7]=prune(count['S'],"SEVEN");
        ans[9]=prune(count['I'],"NINE");
        
        isempty();
        
        for (int i=0;i<10;i++) {
            for (int j=0;j<ans[i];j++)
                digits.append((char)(i+48));
        }
        out.printf("%s\n",digits);
    
    }
    
}
/*
"*ZERO", "T*WO", "FO*UR", "SI*X", "EI*GHT", "*ONE",  "T*HREE",  
"*FIVE",  "*SEVEN", , "N*INE"
*/
